### PR TITLE
Register

### PR DESCRIPTION
**Pull request by Arnas277**
Registracija dar neišbaigta, truksta data validation su formom. Bet viskas veikia. Veliau pataisysiu.

### CODE REVIEW

File: src/Controller/RegisterController.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        if ($data['password'] != $data['passwordConfirmation']) {
+            $error = "Passwords are not the same!";
+        } elseif (strlen($data['password']) < 6) {
+            $error = "Your password is to short. Please, try again.";
+        } elseif (isset($username)) {
+            $error = "User with this username already exists. Please enter different username and try again.";
+        } elseif (isset($email)) {
+            $error = "User with this email already exists. Please enter another email and try again, or try to login.";
+        } else {
+            $data['password'] = password_hash($data['password'], PASSWORD_ARGON2I);
```
Comment (zmoginas) [inline src/Controller/RegisterController.php]: vulnerability - type juggling
This hole block must be converted into form

File: src/Controller/RegisterController.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            $user->setRoles('ROLE_USER');
+            $entityManager->persist($user);
+            $entityManager->flush();
+            $encoders = [new XmlEncoder(), new JsonEncoder()];
+            $normalizers = [new ObjectNormalizer()];
+            $serializer = new Serializer($normalizers, $encoders);
+            $user = $serializer->serialize($user, 'json');
+        }
+
+        // Response -------------------------------------------------------
```
Comment (zmoginas) [inline src/Controller/RegisterController.php]: if you are commenting something, please remove before you pushing into to the develop or master

File: src/Controller/RegisterController.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        if (isset($error)) {
+            $response->setContent($error);
+            $response->setStatusCode(Response::HTTP_OK);
+        } else {
+            $response->headers->setCookie(Cookie::create('name', $data['username']));
+            $response->setContent($user);
+            $response->setStatusCode(Response::HTTP_CREATED);
+        }
+        $response->headers->set('Content-Type', 'text/plain');
+        $response->send();
```
Comment (zmoginas) [inline src/Controller/RegisterController.php]: this must be handled by form

File: src/DataFixtures/SkillsFixtures.php [hunk 4]
```diff
@@ -12,7 +12,7 @@ public function load(ObjectManager $manager)
     {
         for ($i = 1; $i < 150; $i++) {
             $skills = new Skills();
-            $skills->setSkills('amazing skill-'.$i);
+            $skills->setSkill('amazing skill-'.$i);
```
Comment (zmoginas) [inline src/DataFixtures/SkillsFixtures.php]: cs (ctrl+alt+L in phpstorm)

File: src/Repository/UsersRepository.php [hunk 5]
```diff
@@ -19,6 +19,10 @@ public function __construct(ManagerRegistry $registry)
         parent::__construct($registry, Users::class);
     }
 
+    public function findAllUsers()
+    {
+        return "va blecha";
```
Comment (zmoginas) [inline src/Repository/UsersRepository.php]: ?

File: src/Controller/RegisterController.php (hunk 3 above)
Comment (zmoginas) [inline src/Controller/RegisterController.php]: This block must be moved to form

File: src/Controller/RegisterController.php (hunk 2 above)
Comment (zmoginas) [inline src/Controller/RegisterController.php]: if you are commenting something, please remove before you pushing into to the develop or master

File: src/DataFixtures/SkillsFixtures.php (hunk 4 above)
Comment (zmoginas) [inline src/DataFixtures/SkillsFixtures.php]: cs (ctrl+alt+L in phpstorm)

File: src/Repository/UsersRepository.php (hunk 5 above)
Comment (zmoginas) [inline src/Repository/UsersRepository.php]: ?

File: assets/js/components/layoutUserDashboard.js [hunk 6]
```diff
@@ -55,11 +55,11 @@ const LayoutUserDashboard = ({ history, location, children }) => {
         <div className="w-full h-auto my-8 items-center hidden xl:flex">
           <div className="bg-blue-500 h-12 w-12 rounded-full flex justify-center items-center">
             <span className="text-white text-2xl font-semibold">
-              {user.name.split("")[0]}
+              kazkas
```
Comment (Karcikovas) [inline assets/js/components/layoutUserDashboard.js]: Kazkas?

File: src/Repository/UsersRepository.php (hunk 5 above)
Comment (Karcikovas) [reply src/Repository/UsersRepository.php]: "va blecka"

**Pull request merged**